jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -58,18 +58,18 @@
 import javax.annotation.Nullable;
 
 /** Pulls the base image manifests for the specified platforms. */
-class PullBaseImageStep implements Callable<List<ImageAndRegistryClient>> {
+class PullBaseImageStep implements Callable<BaseImagesAndRegistryClient> {
 
   private static final String DESCRIPTION = "Pulling base image manifest";
 
   /** Structure for the result returned by this step. */
-  static class ImageAndRegistryClient {
+  static class BaseImagesAndRegistryClient {
 
-    final Image image;
+    final List<Image> baseImages;
     @Nullable final RegistryClient registryClient;
 
-    ImageAndRegistryClient(Image image, @Nullable RegistryClient registryClient) {
-      this.image = image;
+    BaseImagesAndRegistryClient(List<Image> baseImages, @Nullable RegistryClient registryClient) {
```

Comment:
Although the new name works, I think we can keep this general by `ImagesAndRegistryClient`. It's just a tuple of two things (an `Image` list and a `RegistryClient`) only required in Java to return multiple objects. If we give the name "Base", then other people who don't look into the implementation of this tuple will assume that there must be something specific in the class that has to be tied to base images. As a class, I think the name should just reflect what the class actually represents, not how it must be used with only a subset of semantics. For example, a (bad) analogy would be that I rename a general `Integer` class to `PositiveInteger` while not changing anything about the class implementation only because I will never use the class to store a non-positive integer throughout my application lifecycle. Rather, I should give the `positiveInteger` name to an instance of the `Integer` class (e.g., `Integer positiveInteger = 30;`). Does that make sense?

Reply:
Yes I get what you mean.The images stored in these classes are no different form regular images so there is no need to rename the class.